### PR TITLE
Support fqcn aliasses

### PR DESCRIPTION
**Pull request by brendt**
Addresses #13

### CODE REVIEW

**Comment by brendt** [comment]
I'm not sure we should do this. It's a lot of code for a tiny little feature, which goes against many doc block guidelines. 

Not to mention the impact this has on performance: every value object now must open and scan its class file, that's a lot of file IO..

**Comment by brendt** [comment]
I don't think we want to do this at the moment. There's just way too much overhead.

**Comment by brendt** [comment]
@mfn let's continue our discussion here 🙂

**Comment by brendt** [comment]
There's two issue I already of with this PR:

- `use as` is not supported yet
- every DTO will re-read its source file, even if these DTOs are the same. This becomes highly inefficient in collections.

**Comment by brendt** [comment]
screen shot 2018-11-08 at 10 02 20

Here's another issue with in IDEs. This import statement is regarded as unused..

**Comment by mfn** [comment]
> * every DTO will re-read its source file, even if these DTOs are the same. This becomes highly inefficient in collections.

Would a solution using a private static member to cache this be deemed appropriate? …

…under the following assumption: the source never changes (as in: no one mucks with the already loaded in-memory class, i.e. no runkit-shenanigan, etc.)

**Comment by brendt** [comment]
@mfn static caching is an option. Or we could have a separate cache store, two valid solutions.

File: tests/ValueObject/ReflectionClassTest.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+    /** @test */
+    public function it_parses_use_statements()
+    {
+        $dataTransferObject = new class([]) extends DataTransferObject {
+        };
+
+        $class = new DataTransferObjectDefinition($dataTransferObject);
+
+        $this->assertEquals(DummyClass::class, $class->resolveAlias('DummyClass'));
```
Comment (peter279k) [inline tests/ValueObject/ReflectionClassTest.php]: It should use the `assertInstanceOf` rather than `assertEquals` assertion :).

**Comment by peter279k** [comment]
The `tests/DataTransferTransferObjectTest.php` has conflicts.

It should be fixed.

**Comment by brendt** [comment]
After using the package in a lot of projects, I'm now convinced there's little value in adding this feature. It's at least not worth the performance cost.

**Pull request closed without merging**